src/MBrace.Streams/CloudStreams.fs

```diff
@@ -44,25 +44,16 @@ module CloudStream =
     [<Literal>]
     let private maxCloudVectorPartitionSize = 1073741824L // 1GB
 
-    /// If local context then returns number of cores instead of throwing exception
-    /// else returns number of workers.
-    let inline private getWorkerCount() = cloud {
-        let! ctx = Cloud.GetSchedulingContext()
-        match ctx with
-        | Sequential  -> return failwith "Invalid CloudStream context : %A" ctx
-        | ThreadParallel -> return 1
-        | Distributed -> return! Cloud.GetWorkerCount()
-    }
 
     /// gets all partition indices found in cloud vector
     let inline private getPartitionIndices (v : CloudVector<'T>) = [| 0 .. v.PartitionCount - 1 |]
 
     /// Flat map/reduce with sequential execution on leafs.
-    let inline private parallelInChunks (npar : int) (workflows : Cloud<'T> []) = cloud {
+    let inline private parallelInChunks (npar : int) (workflows : Local<'T> []) = cloud {
         let! partials = 
             workflows 
             |> Partitions.ofArray npar
-            |> Array.map (fun wfs -> wfs |> Cloud.Parallel |> Cloud.ToSequential)
+            |> Array.map (fun wfs -> wfs |> Cloud.LocalParallel)
```

Comment:
You have a point, but in any case the function parallelInChunks is never used and can be deleted.